app/controllers/api/account.php

```diff
@@ -476,10 +479,15 @@
             }
         }
 
-        $user = ($user->isEmpty()) ? $dbForProject->findOne('sessions', [ // Get user by provider id
-            Query::equal('provider', [$provider]),
-            Query::equal('providerUid', [$oauth2ID]),
-        ]) : $user;
+        if ($user->isEmpty()) {
+            $session = $dbForProject->findOne('sessions', [ // Get user by provider id
+                Query::equal('provider', [$provider]),
+                Query::equal('providerUid', [$oauth2ID]),
+            ]);
+            if ($session !== false && !$session->isEmpty()) {
+                $user->setAttributes($dbForProject->getDocument('users', $session->getAttribute('userId'))->getArrayCopy());
+            }
+        }
```

Comment:
Should we really try to look up a user by provider? That sounds dangerous...Maybe we should delete this block altogether.

Reply:
What exactly is this part responsible for?

Reply:
This is the old code that (I assume) tried to find the user from an old session with matching provider values:

https://github.com/appwrite/appwrite/blob/c8689d1d416716e74a47c95957f96d15ea23baf0/app/controllers/api/account.php#L479-L482

However, this never worked because:

1. an unauthenticated user doesn't have access to another user's sessions
2. assigning a session to $user wouldn't work